sbin/mount_cd9660/mount_cd9660.c

```diff
@@ -254,3 +272,58 @@ set_charset(struct iovec **iov, int *iovlen, const char *localcs)
 
 	return (0);
 }
+
+static gid_t
+a_gid(const char *s)
+{
+	struct group *gr;
+	const char *gname;
+	gid_t gid;
+
+	if ((gr = getgrnam(s)) != NULL)
+		gid = gr->gr_gid;
+	else {
+		for (gname = s; *s && isdigit(*s); ++s);
+		if (!*s)
+			gid = atoi(gname);
+		else
+			errx(EX_NOUSER, "unknown group id: %s", gname);
+	}
+	return (gid);
+}
+
+static uid_t
+a_uid(const char *s)
+{
+	struct passwd *pw;
+	const char *uname;
+	uid_t uid;
+
+	if ((pw = getpwnam(s)) != NULL)
+		uid = pw->pw_uid;
+	else {
+		for (uname = s; *s && isdigit(*s); ++s);
+		if (!*s)
+			uid = atoi(uname);
+		else
+			errx(EX_NOUSER, "unknown user id: %s", uname);
+	}
+	return (uid);
+}
+
+static mode_t
+a_mask(const char *s)
+{
+	int done, rv;
+	char *ep;
+
+	done = 0;
```

Comment:
You might be able to simplify this slightly as:

```
     rv = strtol(optarg, &ep, 8);
     if (ep == optarg || rv < 0 || *ep != '\0')
         errx(...)
```

That is, relying on strtol to return optarg == ep if the first character is invalid and not needing `done` or a default value for `rv`.

Note that FreeBSD's style prefers explicit tests for things like `foo != NULL` or `bar != '\0'` rather than relying on implicit tests like `if (foo)`.

Reply:
I copied this verbatim from `mount_msdosfs.c`:

https://github.com/freebsd/freebsd-src/blob/main/sbin/mount_msdosfs/mount_msdosfs.c#L277

Reply:
That's fair.